docs/guide/creating_a_gem.md

```diff
@@ -0,0 +1,148 @@
+---
+layout: default
+title: Creating a Gem
```

Comment:
Do you think this document should go under Guide? Or maybe a new parent heading?

Reply:
What about putting it under _Practical Examples_ as introduced in #946?

Reply:
@Spone I'm torn about adding that new section. Maybe we need to take step back and think about our docs organization more broadly.

Reply:
Feel free to reach out if you'd like to talk about it (or open a new issue for that?)

Reply:
It might be nice to have a group chat about this with @coderberry and whoever else might be interested. I know that he had some thoughts about higher-level docs goals.